amqpdispatcher/channel_proxy.py

```diff
@@ -54,6 +54,32 @@ def queue_declare(self, callback=None, queue='', passive=False,
         consumer_count = ret.method.consumer_count
         return name, message_count, consumer_count
 
+    def basic_consume(self,
+                      consumer_callback=None,
+                      queue='',
+                      no_ack=False,
+                      exclusive=False,
+                      consumer_tag=None,
+                      arguments=None):
+        if arguments is None:
+            arguments = {}
+
+        no_local = arguments.get('no_local', False)
+        nowait = arguments.get('nowait', True)
+        ticket = arguments.get('ticket', None)
+        cb = arguments.get('cb', None)
+        if consumer_tag is None:
+            consumer_tag = ''
+        return self._channel.basic.consume(queue=queue,
+                                           on_message_callback=consumer_callback,
+                                           no_ack=no_ack,
+                                           exclusive=exclusive,
+                                           consumer_tag=consumer_tag,
+                                           no_local=no_local,
+                                           nowait=nowait,
```

Comment:
i think a couple of these should be under arguments as a dict

Reply:
I actually disagree here. Overusing *args or **kwargs can tend to be an anti-pattern in some cases. In most cases, it should be obvious what a method signature is by just glancing.

Reply:
i was referring to the underlying pika method, it takes in `arguments` that is a dict, rather than what haigha did which is what you were describing